src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -595,7 +579,7 @@ private bool ValidatePackagesSha512(PackagesLockFile lockFile, LockFile assetsFi
                         // bail restore since it's the locked mode but required to update the lock file.
                         // directly log to the request logger when we're not going to rewrite the assets file otherwise this log will
                         // be skipped for netcore projects.
-                        await _request.Log.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1004, Strings.Error_RestoreInLockedMode));
+                        await _logger.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1004, Strings.Error_RestoreInLockedMode));
```

Comment:
Here you changed the logger, but line#580 it states `// directly log to the request logger when we're not going to rewrite the assets file otherwise this log will`.
So now is this _logger (instance of RestoreCollectorLogger) actually writing anything to lock file or asset file?